app/routes/list-project.tsx

```diff
@@ -0,0 +1,531 @@
+import TwitterIcon from "~/components/icons/TwitterIcon";
+import { LinkIcon, MarkGithubIcon } from "@primer/octicons-react";
+import ProjectPreview from "~/components/ProjectPreview";
+import { Project } from "~/types";
+import ProjectSubmissionSpinner from "~/components/ProjectSubmissionSpinner";
+import RequiredMarker from "~/components/RequiredMarker";
+import ExternalLink from "~/components/ExternalLink";
+import mapThumbnailSrc from "~/images/map_thumbnail.png";
+import { formatFileSize, maybeStringToArray } from "~/utils/formatting";
+import FieldError from "~/components/FieldError";
+import {
+  ActionFunction,
+  json,
+  LoaderFunction,
+  redirect,
+} from "@remix-run/node";
+import {
+  getAccessToken,
+  getCurrentUser,
+  getCurrentUserOrRedirect,
+} from "~/session.server";
+import {
+  MAX_AVATAR_SIZE_BYTES,
+  parseListProjectForm,
+} from "~/utils/project-submission";
+import { createNewPullRequest } from "~/github.server";
+import { getRepoOwnerAndName } from "~/utils/repo-url";
+
+export function links() {
+  return [
+    { rel: "stylesheet", href: formStyles },
+    { rel: "stylesheet", href: markdownStyles },
+  ];
+}
+
+/**
+ * Check that the user is logged in or redirect to the /login page
+ */
+export const loader: LoaderFunction = async ({ request }) => {
+  await getCurrentUserOrRedirect(request);
+  return json({});
+};
+
+/**
+ * This method is called when the form is submitted. It performs data validation
+ * and uploads files to GitHub before creating a PR. If all goes well, it
+ * redirects to a success page.
+ */
+export const action: ActionFunction = async ({ request }) => {
+  const currentUser = await getCurrentUser(request);
+  const accessToken = await getAccessToken(request);
+
+  if (!currentUser || !accessToken) {
+    throw new Error("You must be logged in to save your profile");
+  }
+
+  // TODO check whether we already have a project for this repository
+
+  // Parse the form
+  const { validationErrors, files, repoUrl } = await parseListProjectForm(
+    request,
+    currentUser
+  );
+
+  // If there are any validation errors, return early
+  if (validationErrors) {
+    return json({ validationErrors });
+  }
+
+  if (!repoUrl) {
+    throw new Error("Missing repo URL in new project form");
+  }
+
+  // Create a new pull request with the content of the form
+  const { owner: repoOwner, name: repoName } = getRepoOwnerAndName(repoUrl);
+  const { pullRequestUrl } = await createNewPullRequest(
+    accessToken,
+    files,
+    repoOwner,
+    repoName
+  );
+
+  return redirect(`/project-listed?pr=${encodeURIComponent(pullRequestUrl)}`);
+};
+
+type TagsState = {
+  languages: MultiValue<{ label: string; value: string }>;
+  currentlySeeking: MultiValue<{ label: string; value: string }>;
+  tags: MultiValue<{ label: string; value: string }>;
+};
+
+/**
+ * This (massive) form allows users to list their projects on Open Source Hub.
+ */
+const ListProject: FC = () => {
```

Comment:
ℹ️ This is the start of a Code Review Tour! [Check it out here]https://app.codesee.io/maps/review/github/Codesee-io/opensourcehub/pr/110).

This new file contains the client and server code for the new form. I tried to use the vanilla form functionality as much as possible, but I relied on a few `useState` calls for the more advanced functionality like the avatar upload and lists of tags.